### PR TITLE
Fix service cluster plugin bug

### PR DESCRIPTION
**Pull request by arugal**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
#3069 
___

All fixed and added IT tests

### CODE REVIEW

**Comment by arugal** [comment]
@wu-sheng please review

File: oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            config.setInternalComPort(internalComPort);
+        }
+
+        provider.prepare();
+        provider.start();
+        provider.notifyAfterCompleted();
+
+        ConsulCoordinator consulCoordinator = (ConsulCoordinator) provider.getService(ClusterRegister.class);
+        // ignore health check
+        ClusterRegister register = new ClusterRegister() {
```
Comment (kezhenxu94) [inline oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java]: Replace with lambda

File: oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author: zhangwei
```
Comment (kezhenxu94) [inline oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java]: Remove all the colons(`:`) after `@author`

File: oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java (hunk 1 above)
Comment (arugal) [reply oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java]: ```remoteInstance``` needs to be reassigned

Comment (kezhenxu94) [reply oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java]: > `remoteInstance` needs to be reassigned

Doesn't it compile? It works well on my machine
image

Comment (arugal) [reply oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java]: done

File: oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java (hunk 2 above)
Comment (arugal) [reply oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java]: done

**Comment by wu-sheng** [comment]
@arugal I have changed the title of this PR, please add a clear title next time, to benefit search.

**Comment by arugal** [comment]
> look like only Zookeeper coordinator needs to change for receiver mode

```Zookeeper Nacos``` and ```Consul```

**Comment by wu-sheng** [comment]
Such as, what is the bug of `NacosCoordinator`? I know you change the codes, but it doesn't seem like bug, more like just code refactor.  Do I miss anything?

**Comment by arugal** [comment]
https://github.com/apache/skywalking/blob/1d13d735bbc1db1edd5a70b92b6bffd40bd9e4cc/oap-server/server-cluster-plugin/cluster-nacos-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/nacos/NacosCoordinator.java#L48-L65
When the node is set to ```Receiver``` mode, ```#registerRemote``` is not invoker, ```selfAddress``` is null, right?

**Comment by wu-sheng** [comment]
Got your point. Make sense. Waiting for CI to merge.

**Comment by arugal** [comment]
I missed the IT test of k8s

**Comment by wu-sheng** [comment]
> I missed the IT test of k8s

I know. We need k8s e2e test as a higher priority, if you have interests, we could discuss that.

**Comment by arugal** [comment]
> We need k8s e2e test as a higher priority

So I don't need to add IT test of k8s for the time being ?

> if you have interests, we could discuss that.

Yes

**Comment by wu-sheng** [comment]
> So I don't need to add IT test of k8s for the time being ?

For now, yes. You don't need to do that.

> if you have interests, we could discuss that. Yes

Do you have my personal mail/wechat/qq group? Ping me there. I could set up a discussion group for k8s e2e test

**Comment by wu-sheng** [comment]
Check the e2e, whether this change breaks test. We test e2e w/ zookeeper

**Pull request merged**